src/lib/Libifl/int_manager.c

```diff
@@ -86,31 +103,58 @@ PBSD_manager(int c, int function, int command, int objtype, char *objname, struc
 			return pbs_errno;
 
 	/* now verify the attributes, if verification is enabled */
-	if ((pbs_verify_attributes(c, function, objtype,
-		command, aoplp)) != 0)
-		return pbs_errno;
-
-	/* lock pthread mutex here for this connection */
-	/* blocking call, waits for mutex release */
-	if (pbs_client_thread_lock_connection(c) != 0)
+	if ((pbs_verify_attributes(c, function, objtype, command, (struct attropl *) aoplp)))
 		return pbs_errno;
 
-	/* send the manage request */
-	i = PBSD_mgr_put(c, function, command, objtype, objname, aoplp, extend, PROT_TCP, NULL);
-	if (i) {
-		(void)pbs_client_thread_unlock_connection(c);
-		return i;
+	if ((objtype == MGR_OBJ_JOB) && (get_server(objname, job_id_out, server_out) == 0)) {
+		if (PBS_get_server(server_out, server_name, &server_port)) {
+			single_itr = 1;
+			for (i = 0; i < num_cfg_svrs; i++) {
+				if (!strcmp(server_name, pbs_conf.psi[i].name) && server_port == pbs_conf.psi[i].port) {
+					start = i;
+					break;
+				}
+			}
+		}
 	}
 
-	/* read reply from stream into presentation element */
-	reply = PBSD_rdrpy(c);
-	PBSD_FreeReply(reply);
+	for (i = start, ct = 0; ct < num_cfg_svrs; i = (i + 1) % num_cfg_svrs, ct++) {
 
-	rc = get_conn_errno(c);
+		if (svr_connections[i].state != SVR_CONN_STATE_UP) {
+			pbs_errno = PBSE_NOSERVER;
+			agg_rc = pbs_errno;
+			continue;
+		}
 
-	/* unlock the thread lock and update the thread context data */
-	if (pbs_client_thread_unlock_connection(c) != 0)
-		return pbs_errno;
+		c = svr_connections[i].sd;
+
+		/* lock pthread mutex here for this connection */
+		/* blocking call, waits for mutex release */
+		if (pbs_client_thread_lock_connection(c) != 0)
+			return pbs_errno;
+
+		/* send the manage request */
+		rc = PBSD_mgr_put(c, function, command, objtype, objname, aoplp, extend, PROT_TCP, NULL);
+		if (rc) {
+			pbs_client_thread_unlock_connection(c);
+			return rc;
+		}
+
+		/* read reply from stream into presentation element */
+		reply = PBSD_rdrpy(c);
+		PBSD_FreeReply(reply);
+
+		rc = get_conn_errno(c);
+		if (rc != 0)
+			agg_rc = rc;
```

Comment:
We will bail out if we cannot reach one of the servers. Which is handled in make_connection function within qmgr itself. This is a slightly bigger issue. The command can fail due to internal server error such as cannot delete an object which is busy etc. This will be handled in the future with two-phase commit or so.

Right now the admin will get an error message on the failure and had to make sure that objects are in sync.